### ISSUE TITLE
xcodebuild crashes but swiftc does not

### ISSUE THREAD

**Issue by cortices** [practicalswift/swift-compiler-crashes#104]
This project https://github.com/ObjSer/objser-swift crashes xcodebuild when it is built with the latest swift snapshot (2.2, 2016-01-25), but when the culprit file (in this case IntegralType.swift) is copied into a single swift file along with the blank protocol declaration it relies on in NumericType.swift, it compiles fine with swiftc. The output from the error is as follows:

~~~~
<...>
CompileSwift normal x86_64 /<...>/objser-swift/ObjSer/util/IntegralType.swift
<...>
Assertion failed: (SI->getNumOperands() == STy->getNumElements() && "mismatch StructInst with its lowered StructType!"), function getConstantValue, file /Users/buildnode/jenkins/workspace/oss-swift-2.2-package-osx/swift/lib/IRGen/IRGenSIL.cpp, line 4622.
0  swift                    0x000000010ac3d90b llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 43
1  swift                    0x000000010ac3cbc6 llvm::sys::RunSignalHandlers() + 70
2  swift                    0x000000010ac3dfb2 SignalHandler(int) + 322
3  libsystem_platform.dylib 0x00007fff8da1952a _sigtramp + 26
4  swift                    0x000000010b699c05 FirstTarget + 69373
5  swift                    0x000000010ac3ddc6 abort + 22
6  swift                    0x000000010ac3dda1 __assert_rtn + 81
7  swift                    0x000000010865cfc5 getConstantValue(swift::irgen::IRGenModule&, llvm::StructType*, swift::StructInst*) + 1109
8  swift                    0x000000010865cac5 swift::irgen::IRGenModule::emitSILStaticInitializer() + 245
9  swift                    0x0000000108597449 swift::irgen::IRGenModuleDispatcher::emitGlobalTopLevel() + 553
10 swift                    0x000000010863e616 performIRGeneration(swift::IRGenOptions&, swift::ModuleDecl*, swift::SILModule*, llvm::StringRef, llvm::LLVMContext&, swift::SourceFile*, unsigned int) + 1110
11 swift                    0x000000010863ebb6 swift::performIRGeneration(swift::IRGenOptions&, swift::SourceFile&, swift::SILModule*, llvm::StringRef, llvm::LLVMContext&, unsigned int) + 70
12 swift                    0x00000001084fc812 frontend_main(llvm::ArrayRef<char const*>, char const*, void*) + 11890
13 swift                    0x00000001084f5a49 main + 2905
14 libdyld.dylib            0x00007fff8b8065ad start + 1
15 libdyld.dylib            0x000000000000004b start + 1954519711
Stack dump:
0.	Program arguments: <arguments omitted>
~~~~

From what is printed, it seems the stack dump isn't working either... I would have expected program arguments to come afterwards on its own line.

Any ideas how this could be reliably crashing xcodebuild?

**Comment by practicalswift** [comment]
@cortices Sorry for the late reply! Does this issue still crash the latest Swift snapshot?

**Comment by cortices** [comment]
No, it was solved in the Xcode 7.3 release I believe, but I had forgotten about opening this issue.
